packages/babel-plugin-ember-test-metadata/utils.js

```diff
@@ -85,13 +85,54 @@ function _getParsedEmbroiderFilepath(pathSegments) {
  * @returns {string} E.g. tests/acceptance/my-test.js
  */
 function getNormalizedFilePath(fileOpts, projectConfiguration) {
-  let { root, filename } = fileOpts;
+  let { filename, root } = fileOpts;
   const pathSegments = filename.split(path.sep);
   const isEmbroider = pathSegments.includes('embroider');
-
-    filename = isEmbroider ? _getParsedEmbroiderFilepath(pathSegments) : _getParsedClassicFilepath(pathSegments, projectConfiguration);
-
-  return path.relative(root, filename);
+  const addons = projectConfiguration.pkg["ember-addon"];
+  const hasAddons = addons && addons.paths;
+
+  if (!isEmbroider && !hasAddons) {
+    const projectNamePathSeparators = projectConfiguration.pkg.name.split(path.sep);
+    pathSegments.splice(
+      0,
+      pathSegments.lastIndexOf(root) + projectNamePathSeparators.length
+    );
+
+    return pathSegments.join(path.sep);
+  } else if (!isEmbroider && hasAddons) {
+    const addonNames = addons.paths.map(addonPath => addonPath.split(path.sep)[1]);
+    const addonName = addonNames.find(name => pathSegments.includes(name));
+
+    if (pathSegments.includes(addonName)) {
+      pathSegments.splice(0, pathSegments.lastIndexOf(addonName) + 1);
+      return ['lib', addonName, "tests", ...pathSegments].join(path.sep)
+    } else {
+      const projectNamePathSeparators = projectConfiguration.pkg.name.split(path.sep);
+      pathSegments.splice(
+        0,
+        pathSegments.lastIndexOf(root) + projectNamePathSeparators.length
+      );
+      return pathSegments.join(path.sep);
+    }
```

Comment:
I think each of these should be separate utils, or normalizers. That way we can compose the normalizers based on what we need to support.